src/clojure/pantomime/mime.clj

```diff
@@ -84,3 +86,34 @@
   (when-not (nil? mime-adder)
     (let [mime-type (.forName mime-adder name)]
       (.addPattern mime-adder mime-type pattern true))))
+
+;; I would love to know why there are two separate, completely
+;; independent mechanisms for representing these kinds of objects. -djt
+
+;; Nevertheless, coercions are handy for switching between them.
+
+(defprotocol MimeTypeCoercions
+  (as-mime-type  [x] "Turn a { String, MediaType } object into a MimeType.")
+  (as-media-type [x] "Turn a { String, MimeType } object into a MediaType.")
+)
+
+(extend-protocol MimeTypeCoercions
+  String
+  ;; XXX exception behaviour for these two is an architecture decision. -djt
+  (as-mime-type  [^String x] (try (for-name x)))
+  (as-media-type [^String x] (try (.getType (as-mime-type x))))
+  MimeType
+  (as-mime-type  [^MimeType x] x)
+  (as-media-type [^MimeType x] (.getType x))
+  MediaType
+  (as-mime-type  [^MediaType x] (for-name (.toString (.getBaseType x))))
+  (as-media-type [^MediaType x] x)
+)
+
+;; don't collide with 'core/instance?'
+(defn type-instance?
```

Comment:
I'd use `instance-of?`. For some reason "type-instance?" doesn't make me think of MIME types and inheritance.